rfc/0076-subdoc-replica-reads.md

```diff
@@ -0,0 +1,179 @@
+In addition, the SDK MUST broadcast the requests simultaneously to ensure the application receives data as quickly as possible.
+
+```
+Stream<ILookupInReplicaResult> LookupInAllReplicas(string id, LookupInSpec[] specs, [LookupInAllReplicasOptions])
+```
+
+Parameters
+
+- Required
+
+    - Id: string - the primary key.
+    - LookupInSpec - an array of fetch operations, see [LookupIn](./0053-sdk3-crud.md#LookupIn)
+
+- Optional
+    - Timeout or timeoutMillis (int/duration) - the time allowed for the operation to be terminated. This is controlled by the client.
+    - Serializer - a custom serializer for converting the memcached JSON data to a native type.
+
+Returns
+
+A stream of replicas as a ILookupInReplicaResult object
+
+Throws
+
+- Documented
+
+    - RequestTimeoutException (#1)
+    - CouchbaseException (#0)
+    - FeatureNotAvailableException (#15)
+
+- Undocumented
+    - InvalidArgumentException (#3)
+
+## LookupInAnyReplica
+
+Gets a document for a given id, leveraging both the active and all available replicas.
+This method follows the same semantics of LookupInAllReplicas (including the fetch from ACTIVE), but returns the first response as opposed to returning all responses.
+
+Signature
+
+```
+ILookupInReplicaResult LookupInAnyReplica(string id, LookupInSpec[] specs, [LookupInAnyReplicaOptions])
+```
+
+Parameters
+
+- Required
+
+    - Id: string - the primary key.
+    - LookupInSpec - an array of fetch operations, see [LookupIn](./0053-sdk3-crud.md#LookupIn)
+
+- Optional
+
+    - Timeout or timeoutMillis (int/duration) - the time allowed for the operation to be terminated. This is controlled by the client.
+    - Serializer - a custom serializer for converting the memcached JSON data to a native type.
+
+Returns
+
+- A ILookupInReplicaResult wrapping a ILookupInResult object.
+
+Throws
+
+NOTE: The following exceptions are thrown at the "top" level. The errors that are thrown on a per-field basis are covered in [LookupIn](./0053-sdk3-crud.md#LookupIn).
+
+- Throws
+    - Documented
+        - DocumentUnretrievableException (#125)
+            - The situation where we got responses from all of them (most likely: key not found) but none of them were SUCCESS so we ended up not returning anything (empty stream from lookupin all replicas).
+        - RequestTimeoutException (#1)
+        - CouchbaseException (#0)
+        - FeatureNotAvailableException (#15)
+    - Undocumented
+        - InvalidArgumentException (#3)
+
+## ILookupInReplicaResult
+
+The ILookupInReplicaResult interface is the return type for Sub-Document read operations from replicas.
+
+```csharp
+public interface ILookupInResult : IResult {
+  T ContentAs<T>(int lookupIndex);
+  Boolean Exists(int lookupIndex);
+  Boolean IsReplica();
+}
+```
+
+See [LookupIn](./0053-sdk3-crud.md#LookupIn) for information on `Exists` and `ContentAs`.
+
+- IsReplica
+  - Returns true if the read was from a replica node.
+
+## BucketCapabilities Check
```

Comment:
As of writing this does not actually exist, https://issues.couchbase.com/browse/MB-57470